spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/FirestoreTemplate.java

```diff
@@ -76,7 +75,9 @@
 
 	private final String databasePath;
 
-	private final FirestoreMappingContext mappingContext = new FirestoreMappingContext();
+	private final FirestoreClassMapper classMapper;
+
+	private final FirestoreMappingContext mappingContext;
```

Comment:
It seems odd that you would need both `classMapper` and `mappingContext`. I think only one should be required.

Reply:
I thought about only having a mapping context, but it seemed awkward to add boilerplate of `this.mappingContext.getClassMapper().doWhatever` everywhere.
I have no problem with doing that, though.